components/sup/src/http_gateway.rs

```diff
@@ -198,6 +265,25 @@ impl Server {
     }
 }
 
+fn tls_config(tls_files: Option<(path::PathBuf, path::PathBuf)>) -> Option<ServerConfig> {
+    match tls_files {
+        Some((key_file, cert_file)) => {
+            let mut config = ServerConfig::new(NoClientAuth::new());
+            let key_file = &mut BufReader::new(File::open(key_file).unwrap());
+            let cert_file = &mut BufReader::new(File::open(cert_file).unwrap());
+
+            // The validation for both the key file and the cert file happen upfront when we parse
+            // CLI args. By the time we get to this place, we know for certain that we have
+            // valid key and cert files, so we unwrap everything here.
+            let cert_chain = certs(cert_file).unwrap();
+            let mut keys = rsa_private_keys(key_file).unwrap();
+            config.set_single_cert(cert_chain, keys.remove(0)).unwrap();
+            Some(config)
+        }
+        None => None,
+    }
+}
```

Comment:
If we make the changes I suggested above, this function simplifies nicely:
```rust
fn tls_config(
    key_file: path::PathBuf,
    cert_file: path::PathBuf,
) -> result::Result<ServerConfig, rustls::TLSError> {
    let mut config = ServerConfig::new(NoClientAuth::new());
    let key_file = &mut BufReader::new(File::open(key_file).unwrap());
    let cert_file = &mut BufReader::new(File::open(cert_file).unwrap());

    // …
    let cert_chain = certs(cert_file).unwrap();
    let mut keys = rsa_private_keys(key_file).unwrap();
    config
        .set_single_cert(cert_chain, keys.remove(0))
        .map(|_| config)
}
```

However, there are still some issues I've noted above that I think need to be resolved.